### ISSUE TITLE
Add accessible titles and aria-hidden support to React icon components

### ISSUE THREAD

**Issue by mkernohanbc** [bcgov/design-system#436]
For commonly-used SVG icons that we implement (or have already implemented) as React components, we should explore adding support for:

- [the `<title>` attribute](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title)
- `aria-hidden` and/or `aria-labelledby`

Without these, I suspect we risk allowing accessibility violations to creep in as we reuse these icon components.

In terms of basic implementation, I think this looks something like:

- add `title` prop to populate `<title>` attribute on the `<svg>` element
- fallback to `aria-hidden="true"` if no title is provided

@ty2k I'm happy to take a first pass at implementing this myself, but would appreciate your thoughts on whether my assumptions here are valid?

**Comment by mkernohanbc** [comment]
POC for this fix: https://github.com/bcgov/design-system/compare/main...436-svg-accessible-titles-aria-hidden
